analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt

```diff
@@ -224,29 +223,31 @@ class MainActivity : AppCompatActivity() {
     }
 
     /**
-     * A [FragmentPagerAdapter] that returns a fragment corresponding to
+     * A [FragmentStateAdapter] that returns a fragment corresponding to
      * one of the sections/tabs/pages.
      */
     @SuppressLint("WrongConstant")
     inner class ImagePagerAdapter(
-        fm: FragmentManager,
+        fm: FragmentActivity,
         private val infos: Array<ImageInfo>
-    ) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
-
-        override fun getItem(position: Int): Fragment {
-            val info = infos[position]
-            return ImageFragment.newInstance(info.image)
-        }
+    ) : FragmentStateAdapter(fm) {
 
-        override fun getCount() = infos.size
-
-        override fun getPageTitle(position: Int): CharSequence? {
+        fun getPageTitle(position: Int): CharSequence? {
             if (position < 0 || position >= infos.size) {
                 return null
             }
             val l = Locale.getDefault()
             val info = infos[position]
-            return getString(info.title).toUpperCase(l)
+            return getString(info.title).uppercase(l)
+        }
+
+        override fun getItemCount(): Int {
+            return infos.size
+        }
```

Comment:
nit: this can actually become a single line:
```suggestion
        override fun getItemCount(): Int = infos.size
```